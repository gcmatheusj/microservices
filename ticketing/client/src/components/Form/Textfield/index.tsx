import { InputHTMLAttributes } from 'react'

interface Props extends InputHTMLAttributes<HTMLInputElement> {
  label: string
}

const Textfield: React.FC<Props> = (props) => (
  <>
    <label
      className="block uppercase tracking-wide text-gray-700 text-xs font-bold mb-2"
      htmlFor={props.id}
    >
      {props.label}
    </label>
    <input
      className="appearance-none block w-full bg-gray-200 text-gray-700 border border-gray-200 rounded py-3 px-4 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
      {...props}
    />
  </>
)

export default Textfield
