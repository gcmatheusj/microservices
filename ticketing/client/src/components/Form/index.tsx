export { default as Textfield } from './Textfield';
