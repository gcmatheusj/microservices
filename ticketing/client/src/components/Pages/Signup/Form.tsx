import { useState, FormEvent } from 'react'

import { Textfield } from 'components/Form'
import Button from 'components/Button'

interface Props {
  handleSubmit: (event: FormEvent<HTMLFormElement>) => void
}

const SignupForm: React.FC<Props> = ({ handleSubmit }) => {
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')

  return (
    <form data-testid="signup-form" onSubmit={handleSubmit}>
      <div className="flex flex-wrap -mx-3 mb-6">
        <div className="w-full px-3">
          <Textfield
            id="email"
            data-testid="signup-input-email"
            label="Email"
            name="email"
            type="email"
            placeholder="Type your email"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
          />
        </div>
      </div>
      <div className="flex flex-wrap -mx-3 mb-6">
        <div className="w-full px-3">
          <Textfield
            id="password"
            data-testid="signup-input-password"
            label="Password"
            name="password"
            type="password"
            placeholder="Type your password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
        </div>
      </div>
      <Button type="submit">Register</Button>
    </form>
  )
}

export default SignupForm
