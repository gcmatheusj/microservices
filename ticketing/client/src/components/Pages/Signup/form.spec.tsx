import { screen, render, fireEvent, act } from '@testing-library/react'

import Form from './Form'

const mockHandleSubmit = jest.fn()

describe('<Signup />', () => {
  it('should render the heading', () => {
    const { container } = render(<Form handleSubmit={mockHandleSubmit} />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should be able to change input values', async () => {
    const { getByTestId } = render(<Form handleSubmit={mockHandleSubmit} />)

    act(() => {
      fireEvent.change(getByTestId('signup-input-email'), {
        target: { value: 'test@test.com' }
      })
      fireEvent.change(getByTestId('signup-input-password'), {
        target: { value: 'pass123' }
      })
    })

    const email = getByTestId('signup-input-email') as HTMLInputElement
    const password = getByTestId('signup-input-password') as HTMLInputElement

    expect(email.value).toBe('test@test.com')
    expect(password.value).toBe('pass123')
  })

  it('should be able to submit form', async () => {
    const { getByTestId } = render(<Form handleSubmit={mockHandleSubmit} />)

    act(() => {
      fireEvent.change(getByTestId('signup-input-email'), {
        target: { value: 'test@test.com' }
      })
      fireEvent.change(getByTestId('signup-input-password'), {
        target: { value: 'pass123' }
      })
    })

    act(() => {
      fireEvent.submit(getByTestId('signup-form'))
    })

    expect(mockHandleSubmit).toHaveBeenCalled()
  })
})
