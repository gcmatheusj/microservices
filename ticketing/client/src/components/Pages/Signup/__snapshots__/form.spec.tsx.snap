// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> should render the heading 1`] = `
<form
  data-testid="signup-form"
>
  <div
    class="flex flex-wrap -mx-3 mb-6"
  >
    <div
      class="w-full px-3"
    >
      <label
        class="block uppercase tracking-wide text-gray-700 text-xs font-bold mb-2"
        for="email"
      >
        Email
      </label>
      <input
        class="appearance-none block w-full bg-gray-200 text-gray-700 border border-gray-200 rounded py-3 px-4 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
        data-testid="signup-input-email"
        id="email"
        label="Email"
        name="email"
        placeholder="Type your email"
        type="email"
        value=""
      />
    </div>
  </div>
  <div
    class="flex flex-wrap -mx-3 mb-6"
  >
    <div
      class="w-full px-3"
    >
      <label
        class="block uppercase tracking-wide text-gray-700 text-xs font-bold mb-2"
        for="password"
      >
        Password
      </label>
      <input
        class="appearance-none block w-full bg-gray-200 text-gray-700 border border-gray-200 rounded py-3 px-4 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
        data-testid="signup-input-password"
        id="password"
        label="Password"
        name="password"
        placeholder="Type your password"
        type="password"
        value=""
      />
    </div>
  </div>
  <button
    class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"
    type="submit"
  >
    Register
  </button>
</form>
`;
